rafx-base/src/resource_map.rs

```diff
@@ -107,6 +107,15 @@ impl ResourceMap {
         }
     }
 
+    pub fn ensure_exists_or_add_default<R>(&mut self)
```

Comment:
I would rename this to `try_insert` with signature 
```rust
pub fn try_insert<R>(
        &mut self,
        r: R,
    ) -> bool
```

Then it would be called like

```rust
        render_resources.insert(MeshRenderNodeSet::default());

        render_resources.try_insert(StaticVisibilityNodeSet::default());
```

The difference being that `insert` would explode on overwrite (as implied by the `TODO` in `insert_by_id`) whereas `try_insert` would only insert the value if it wasn't already present. I think that's a little bit cleaner because it keeps the syntax the same between both calls -- the only difference is the function name.

Reply:
I think for this particular use case (multiple plugins wanting to set the value to something) we should only allow creating the default object. This way the order that plugins are created doesn't affect whether we insert what the mesh plugin's desired state or the sprite plugin's desired state. Maybe in some circumstance we would want that, in which case the plugin should probably insert and verify nothing else was inserted first. Also it's possible an end-user might pre-populate render resources with an overridden instance of type T as a way of configuring something. (Bevy uses this for configuring the task system for example.)

I like the try_insert(), so maybe we could to try_insert_default(), where it inserts T::default() rather than taking a T. (well, R I guess :D)

Reply:
`try_insert_default<T>` sounds good.